tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -316,6 +316,40 @@ public void testInitWithInvalidPackage() throws Exception {
         runMainFunction(projectPath, generatedBalx.toString());
     }
 
+    @Test(description = "Test building a project with a single bal file using the target path")
+    public void testBuildWithTarget() throws Exception {
+        // Test ballerina init
+        ServerInstance ballerinaServer = createNewBallerinaServer();
+        Path projectPath = tempProjectDirectory.resolve("testPackageWithTarget");
+        Path genExecPath = tempProjectDirectory.resolve("generatedExec");
+        Files.createDirectories(projectPath);
+        Files.createDirectories(genExecPath);
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "m\n", "\n", "f"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Path sourceFilePath = projectPath.resolve("main.bal");
+        Assert.assertTrue(Files.exists(sourceFilePath));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+
+        // Remove the .ballerina inside the project because we only want the bal file
+        Files.deleteIfExists(projectPath.resolve(".ballerina").resolve(".gitignore"));
+        Files.deleteIfExists(projectPath.resolve(".ballerina"));
+
+        // Test ballerina build
+        String[] clientArgsForBuild = {"main.bal", "-o", genExecPath.resolve("foo.bal").toString()};
+        ballerinaServer = createNewBallerinaServer();
```

Comment:
Do we have to create a new server instance here?

Reply:
yes the server instance created is deleted after executing a command as in https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/ServerInstance.java#L345

Reply:
I think we have to revisit how we have written tools integration test cases. We have followed the approach of unzip only once and run all the test cases for the ballerina integration test module. We have to follow the same here in order to reduce time with each test cases.